src/effects/VST/VSTEffect.cpp

```diff
@@ -1236,7 +1236,14 @@ bool VSTEffectInstance::RealtimeProcessStart(MessagePackage& package)
    auto& message = static_cast<VSTEffectMessage&>(*package.pMessage);
 
    auto &chunk = message.mChunk;
-   if (!chunk.empty()) {
+
+   // Do not set the chunk if the plugin is by Melda and we are called
+   // in the audio thread - otherwise the whole app will freeze.
+   const bool IsMeldaPlugin = (mVendor == "MeldaProduction");
+   const bool IsAudioThread = (mMainThreadId != std::this_thread::get_id());
+   
+   if (!chunk.empty() && !(IsMeldaPlugin && IsAudioThread) )
```

Comment:
I don’t like this hack.

changing between presets while playing and in plain UI — I predict, will stop working.

Did you look for possibilities of deadlock on a mutex it critical section?  Did you determine whether callSetChunk calls back to AudioMaster?

Our understanding of the problem is not good enough.

Reply:
Another possibility:  power the effect off if it was on.  Set the chunk.  Power it on again if it was on.

Make no test of the vendor.  See if TAL reverb still works correctly that way.

Reply:
You are correct, changing presets while playing and in plain UI stops working with this - the sound will be updated but not the slider positions.

I determined that callSetChunk does call back to AudioMaster, but it does so in an unrepeatable way, please see my comments on the issue for details.

I did not look for possibilities of a deadlock on a mutex in a critical section, honestly I have no idea how to do that? callSetChunk places multiple calls to  ::callDispatcher, and inside there there is a lock on a critical section (implemented using `wxCRIT_SECT_LOCKER`), but afaics that critical section is only there.

Reply:
Deadlock might be obvious if you interrupt in the debugger while the program is frozen and view all effect stacks.  See which threads are waiting on sync objects or perhaps spinning in a wait loop.

Reply:
Another possibility.  The scope of mutual exclusions may need to be wider.  The critical section may need to be held across the duration of all the dispatcher calls done as steps of setting the chunk.

To avoid the thread deadlocking itself, change the wxCRIT object to std::recursive_mutex

Reply:
Update: I inserted print statements before ("lock") and after ("release") the locking of the critical section. When the freeze happens, I see a lock but no release (i.e. I do not see two consecutive locks, that's what I was searching for)

Reply:
I also tried the poweroff - setchunk - poweon suggestion, did not work.